kornia/filters/kernels.py

```diff
@@ -641,19 +642,25 @@ def get_motion_kernel2d(kernel_size: int, angle: Union[torch.Tensor, float],
                     [0.1920, 0.3194, 0.0804],
                     [0.2195, 0.0743, 0.0000]])
     """
+
+    device, dtype = _extract_device_dtype([
+        angle if isinstance(angle, torch.Tensor) else None,
+        direction if isinstance(direction, torch.Tensor) else None,
+    ])
+
     if not isinstance(kernel_size, int) or kernel_size % 2 == 0 or kernel_size < 3:
         raise TypeError("ksize must be an odd integer >= than 3")
 
     if not isinstance(angle, torch.Tensor):
-        angle = torch.tensor([angle])
+        angle = torch.tensor([angle], device=device, dtype=torch.float)
```

Comment:
Shouldn't it be as following?
```suggestion
        angle = torch.tensor([angle], device=device, dtype=dtype)
```

Reply:
Same for the rest I think

Reply:
Yeah. But it's also possible that `dtype` is None, right? It happens when neither `angle` nor `direction` is `torch.Tensor`.

Probably change it to
```suggestion
        angle = torch.tensor([angle], device=device, dtype=dtype if dtype else torch.float)
```

What do you think?

Reply:
If I remember right, it will be set to float32 by default when ```dtype==None```.

Reply:
Yes, you are right! 😁